### PR TITLE
Contract Functionality Implemented

### PR DESCRIPTION
**Pull request by anim-13**
In this PR, I had implemented following features:
1. User can create contract for the property.
2. User can edit the contract.
3. User can view the contract. 
4. User can delete contract.

Trello Link : https://trello.com/c/0lncHZZ3

Below are some screenshots:
Screenshot from 2021-03-23 15-23-48
Screenshot from 2021-03-23 15-24-03
Screenshot from 2021-03-23 15-33-12

### CODE REVIEW

File: app/controllers/contracts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  def contract_params
+    params.require(:contract).permit(:start_date, :end_date, :transaction_id, :amount, :transaction_mode, :estate_id)
+  end
+
+  def set_user
+    @user = current_user
+  end
+
+  def set_contract
+    @contract = @user.contracts.find_by_id(params[:id])
```
Comment (kalpana19sharma) [inline app/controllers/contracts_controller.rb:52]: What if contract doesn't exist for given id?

File: app/controllers/estates_controller.rb [hunk 2]
```diff
@@ -20,6 +20,10 @@ def create
       render :new
     end
   end
+
+  def show
+    @estates = Estate.includes(:contract).references(:contract).where('estate_id IS NULL').where.not(estates: {user_id: current_user})
```
Comment (kalpana19sharma) [inline app/controllers/estates_controller.rb]: Can we use scopes here?

File: app/views/contracts/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<h1>Edit Contract</h1>
+<%= render 'contract_form' %>
```
Comment (kalpana19sharma) [inline app/views/contracts/edit.html.erb]: new line

File: app/views/contracts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Fill the Details</h1>
+<%= render 'contract_form' %>
```
Comment (kalpana19sharma) [inline app/views/contracts/new.html.erb]: newline

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
   root to: 'users#new'
   resources :users do
-    resources :estates, except: :show
+    resources :estates 
     resources :addresses, except: :show
+    resources :contracts
```
Comment (kalpana19sharma) [inline config/routes.rb]: Are you using `show` action as well?

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 5 above)
Comment (anim-13) [reply config/routes.rb:6]: yes mam

File: app/controllers/contracts_controller.rb (hunk 1 above)
Comment (anim-13) [reply app/controllers/contracts_controller.rb:52]: That part is handled in view

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**